submissions/Siusarna/port-sniffer/sniffer.js

```
@@ -0,0 +1,139 @@
+               Address must be specified after the parameter.
+               It can be specified as IP address or URL.
+
+    --ports     Parameter which specifies port range.
+               After the parameter two numbers from 0 to 65535 must be specified with a delimiter '-' between them.
+
+               Example: node sniffer.js --host google.com --port 100-2345
+
+    --help     Parameter will show this instructions.
+`;
+
+const validHost = async host => {
+  if (!host) {
+    process.stdout.write(
+      'Invalid host parameter. Try node sniffer.js --help \n'
+    );
+  }
+  try {
+    return await dns
+      .lookup(host)
+      .then(res => {
+        return res.address;
+      })
+      .catch(error => {
+        process.stdout.write(
+          `Invalit host name. ${error}. Try node sniffer.js --help \n`
+        );
+        return 0;
+      });
+  } catch (e) {
+    process.stdout.write(e);
+    return 0;
+  }
+};
+
+const validatorForPort = port => {
+  let validPort = [1, 65535];
+  if (port) {
+    validPort = port.split('-').map(el => Number(el));
+    if (
+      validPort[0] < 0 ||
+      validPort[1] > 65535 ||
+      validPort[2] ||
+      validPort[0] > validPort[1]
+    ) {
+      throw new Error('Invalid port range. See help \n');
+    }
+  } else {
+    process.stdout.write(
+      'No port range specified. Setting to default. For more information try --help \n'
+    );
+  }
+  return validPort;
+};
+
+const checkConnection = async (port, host) => {
+  let flag = false;
+  await new Promise((resolve, reject) => {
+    const socket = new net.Socket();
+    socket.setTimeout(300);
+    socket.on('connect', () => {
+      process.stdout.write('.');
+      flag = true;
+      socket.destroy();
+      resolve();
+    });
+    socket.on('timeout', () => {
+      socket.destroy();
+      reject();
+    });
+    socket.on('error', () => {
+      socket.destroy();
+      reject();
+    });
+    socket.connect(port, host);
+  }).catch(e => e);
```

This error handling will only hides errors (if any). This code will be hard to debug.